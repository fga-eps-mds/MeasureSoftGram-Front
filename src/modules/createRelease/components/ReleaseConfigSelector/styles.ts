import styled from 'styled-components';

export const Header = styled.div`
  margin-bottom: 64px;
  width: 600px;

  & > h1 {
    font-size: 42px;
    font-weight: 400;
  }

  & > p {
    font-size: 16px;
  }
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 45vh;

  justify-content: center;
  align-content: center;
  align-self: center;
`;
