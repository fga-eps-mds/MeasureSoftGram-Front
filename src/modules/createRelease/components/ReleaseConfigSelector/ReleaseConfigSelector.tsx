import React from 'react';
import { Box, Button } from '@mui/material';
import * as Styles from './styles';

export default function ReleaseConfigSelector() {
  return (
    <>
      <Styles.Header>
        <h1>Planejar Release</h1>
      </Styles.Header>
      <Box display="flex" flexDirection="column">
        <Button variant="contained" color="secondary" style={{ marginBottom: '20px' }}>
          Seguir última configuração
        </Button>
        <Button variant="outlined">Alterar configuração</Button>
      </Box>
    </>
  );
}
