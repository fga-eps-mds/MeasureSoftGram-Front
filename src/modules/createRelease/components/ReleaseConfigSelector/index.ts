import ReleaseConfigSelector from './ReleaseConfigSelector';

export default ReleaseConfigSelector;
