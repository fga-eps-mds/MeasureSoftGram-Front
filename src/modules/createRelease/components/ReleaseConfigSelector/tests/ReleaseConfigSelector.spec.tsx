import React from "react";
import { fireEvent, render } from "@testing-library/react";
import ReleaseConfigSelector from "../ReleaseConfigSelector";

const createReleaseContextMock = () => ({
  setUseLastConfig: jest.fn(),
})

jest.mock('@modules/createRelease/context/useCreateRelease', () => ({
  ...jest.requireActual('@modules/createRelease/context/useCreateRelease'),
  useCreateReleaseContext: createReleaseContextMock
}));

describe("<ReleaseConfigSelector />", () => {
  describe("Snapshot", () => {
    it("Deve corresponder ao Snapshot", () => {
      const tree = render(<ReleaseConfigSelector setActiveStep={jest.fn()} />);
      expect(tree).toMatchSnapshot();
    })
  });

  it("Deve chamar o setActiveStep", () => {
    const setActiveStep = jest.fn();
    const { getByText } = render(<ReleaseConfigSelector setActiveStep={setActiveStep} />);
    const editConfigButton = getByText("Alterar configuração");
    expect(editConfigButton).toBeDefined();
    fireEvent.click(editConfigButton);
    expect(setActiveStep).toHaveBeenCalled();
  })

  it("Deve chamar o setUseLastConfig", () => {
    const setActiveStep = jest.fn();
    const { getByText } = render(<ReleaseConfigSelector setActiveStep={setActiveStep} />);
    const followLastConfigButton = getByText("Seguir última configuração");
    expect(followLastConfigButton).toBeDefined();
    fireEvent.click(followLastConfigButton);
    expect(setActiveStep).toHaveBeenCalled();
  })
});
