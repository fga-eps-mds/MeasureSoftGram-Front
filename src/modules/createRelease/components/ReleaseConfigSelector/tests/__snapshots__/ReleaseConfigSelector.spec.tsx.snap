// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReleaseConfigSelector /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Header-sc-d5c683-0 jBBIFg"
      >
        <h1>
          Planejar Release
        </h1>
      </div>
      <div
        class="styles__Body-sc-d5c683-1 qcviw"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-8gvteo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Seguir última configuração
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-8gvteo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Alterar configuração
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Header-sc-d5c683-0 jBBIFg"
    >
      <h1>
        Planejar Release
      </h1>
    </div>
    <div
      class="styles__Body-sc-d5c683-1 qcviw"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-8gvteo-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Seguir última configuração
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-8gvteo-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Alterar configuração
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
